allennlp/tests/models/biattentive_classification_network_test.py

```diff
@@ -30,4 +40,76 @@ def test_mismatching_dimensions_throws_configuration_error(self):
         # the embedding dimension from the text_field_embedder.
         params["model"]["encoder"]["input_size"] = 10
         with pytest.raises(ConfigurationError):
-            Model.from_params(self.vocab, params.pop("model"))
+            Model.from_params(self.vocab, params.get("model"))
+
+    def test_no_elmo_but_set_flags_throws_configuration_error(self):
+        params = Params.from_file(self.param_file)
+        # There is no elmo specified in self.param_file, but set
+        # use_input_elmo and use_integrator_output_elmo to True.
+        # use_input_elmo set to True
+        tmp_params = deepcopy(params)
+        tmp_params["model"]["use_input_elmo"] = True
+        with pytest.raises(ConfigurationError):
+            Model.from_params(self.vocab, tmp_params.get("model"))
+
+        # use_integrator_output_elmo set to True
+        tmp_params = deepcopy(params)
+        tmp_params["model"]["use_input_elmo"] = False
+        tmp_params["model"]["use_integrator_output_elmo"] = True
+        with pytest.raises(ConfigurationError):
+            Model.from_params(self.vocab, tmp_params.get("model"))
+
+        # both use_input_elmo and use_integrator_output_elmo set to True
+        tmp_params = deepcopy(params)
+        tmp_params["model"]["use_input_elmo"] = True
+        tmp_params["model"]["use_integrator_output_elmo"] = True
+        with pytest.raises(ConfigurationError):
+            Model.from_params(self.vocab, tmp_params.get("model"))
+
+    def test_elmo_but_no_set_flags_throws_configuration_error(self):
+        initial_working_dir = os.getcwd()
+        # Change directory to module root.
+        os.chdir(self.MODULE_ROOT)
```

Comment:
this is rather hacky, perhaps we should consider always changing dir to `self.MODULE_ROOT` before every test?

Reply:
Why do we need to change the directory at all?

Reply:
I don't understand why you need to change directory here at all?

Reply:
The check that throws the `ConfigurationError` in the test is in `__init__`, so it's run after the ELMo object is created. `Model.from_params(self.vocab, params.get("model"))` complains when building `Elmo.from_params` that the options file is not where it expects, since it's a relative path in the fixture.

Reply:
Can we just change the path so that we don't have to change directories?

Reply:
unfortunately we can't, since in general we can make no assumptions about where the tests are run from.

Reply:
Oh, right, yeah, I forgot about that.  I noticed yesterday that the SEMPRE files were getting generated at `allennlp/data/` instead of at `data/`, and I wondered why, and this explains it.  Fair enough.